services/webhook/webhook.go

```diff
@@ -100,13 +101,19 @@ func getPayloadBranch(p api.Payloader) string {
 	return ""
 }
 
+// SourceContext represents the source of a webhook action. Repository and/or Owner must be set.
+type SourceContext struct {
+	Repository *repo_model.Repository
+	Owner      *user_model.User
+}
+
 // PrepareWebhook adds special webhook to task queue for given payload.
-func PrepareWebhook(w *webhook_model.Webhook, repo *repo_model.Repository, event webhook_model.HookEventType, p api.Payloader) error {
-	if err := prepareWebhook(w, repo, event, p); err != nil {
+func PrepareWebhook(ctx SourceContext, w *webhook_model.Webhook, event webhook_model.HookEventType, p api.Payloader) error {
+	if err := prepareWebhook(ctx, w, event, p); err != nil {
 		return err
 	}
 
-	go hookQueue.Add(repo.ID)
+	go hookQueue.Add("dummy")
```

Comment:
Could there a meaningful name instead of "dummy"?

Reply:
The `hookQueue` should be replaced with a `ChannelQueue` (or another type). I did not do that in this PR to keep it small. I will create another PR for that.

Reply:
```suggestion
	go hookQueue.Add("dummy") // the hookQueue should be replaced by a ChannelQueue (or another type) in future. So we do not name it now.
```

How about adding a comment? Correct me if I am wrong.